engine/access/rest_api_test.go

```diff
@@ -0,0 +1,142 @@
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// RestAPITestSuite tests that the Access node serves the REST API defined via the OpenApi spec accurately
+type RestAPITestSuite struct {
+	suite.Suite
+	state      *protocol.State
+	snapshot   *protocol.Snapshot
+	epochQuery *protocol.EpochQuery
+	log        zerolog.Logger
+	net        *network.Network
+	request    *module.Requester
+	collClient *accessmock.AccessAPIClient
+	execClient *accessmock.ExecutionAPIClient
+	me         *module.Local
+	chainID    flow.ChainID
+	metrics    *metrics.NoopCollector
+	rpcEng     *rpc.Engine
+
+	// storage
+	blocks       *storagemock.Blocks
+	headers      *storagemock.Headers
+	collections  *storagemock.Collections
+	transactions *storagemock.Transactions
+	receipts     *storagemock.ExecutionReceipts
+}
+
+func (suite *RestAPITestSuite) SetupTest() {
+	suite.log = zerolog.New(os.Stdout)
+	suite.net = new(network.Network)
+	suite.state = new(protocol.State)
+	suite.snapshot = new(protocol.Snapshot)
+
+	suite.epochQuery = new(protocol.EpochQuery)
+	suite.state.On("Sealed").Return(suite.snapshot, nil).Maybe()
+	suite.state.On("Final").Return(suite.snapshot, nil).Maybe()
+	suite.snapshot.On("Epochs").Return(suite.epochQuery).Maybe()
+	suite.blocks = new(storagemock.Blocks)
+	suite.headers = new(storagemock.Headers)
+	suite.transactions = new(storagemock.Transactions)
+	suite.collections = new(storagemock.Collections)
+	suite.receipts = new(storagemock.ExecutionReceipts)
+
+	suite.collClient = new(accessmock.AccessAPIClient)
+	suite.execClient = new(accessmock.ExecutionAPIClient)
+
+	suite.request = new(module.Requester)
+	suite.request.On("EntityByID", mock.Anything, mock.Anything)
+
+	suite.me = new(module.Local)
+
+	accessIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleAccess))
+	suite.me.
+		On("NodeID").
+		Return(accessIdentity.NodeID)
+
+	suite.chainID = flow.Testnet
+	suite.metrics = metrics.NewNoopCollector()
+
+	const anyPort = ":0" // :0 to let the OS pick a free port
+	config := rpc.Config{
+		UnsecureGRPCListenAddr: anyPort,
+		SecureGRPCListenAddr:   anyPort,
+		HTTPListenAddr:         anyPort,
+		RESTListenAddr:         anyPort,
+		ServeRESTAPI:           true,
+	}
+
+	suite.rpcEng = rpc.New(suite.log, suite.state, config, suite.collClient, nil, suite.blocks, suite.headers, suite.collections, suite.transactions,
+		nil, nil, suite.chainID, suite.metrics, 0, 0, false, false, nil, nil)
+	unittest.AssertClosesBefore(suite.T(), suite.rpcEng.Ready(), 2*time.Second)
+
+	// wait for the server to startup
+	assert.Eventually(suite.T(), func() bool {
+		return suite.rpcEng.RestApiAddress() != nil
+	}, 5*time.Second, 10*time.Millisecond)
+}
+
+func TestRestAPI(t *testing.T) {
+	suite.Run(t, new(RestAPITestSuite))
+}
+
+func (suite *RestAPITestSuite) TestRestAPICall() {
+
+	suite.Run("happy path - REST client successfully executed the GetBlockByID request", func() {
```

Comment:
Can we add tests for the two unhappy paths: BadRequest and NotFound

Reply:
added